setup-nextcloud-hpb.sh

```diff
@@ -325,6 +325,25 @@ function log() {
 	echo -e "$@" 2>&1 | tee -a $LOGFILE_PATH
 }
 
+function generate_dhparam_file() {
+	if [ "$BUILT_DHPARAM_FILE" == "true" ]; then
+		# Skip if we already generated dhparam file.
+		return 0
+	fi
+
+	if [ -s "$DHPARAM_PATH" ]; then
+		# Rebuilding dhparam file.
+		rm -fv "$DHPARAM_PATH" 2>&1 | tee -a "$LOGFILE_PATH"
+	fi
+
+	is_dry_run || mkdir -p "$(dirname $DHPARAM_PATH)"
+	is_dry_run || touch "$DHPARAM_PATH"
+	is_dry_run || openssl dhparam -dsaparam -out "$DHPARAM_PATH" 4096
+	is_dry_run || chmod 644 "$DHPARAM_PATH"
```

Comment:
I'm not sure if the DHPARAM file is security-sensitive. @DecaTec do you know that?
Should we protect the file even more or can we be relaxed with file permissions?